File: routes/activities.py
"""
Activity/time tracking routes.

Handles activity CRUD operations for cases.
"""

import asyncio
from fastapi.responses import JSONResponse
from pydantic import ValidationError
import database as db
import auth
from schemas import CreateActivityInput
from .common import api_error, pydantic_error


def register_activity_routes(mcp):
    """Register activity management routes."""

    @mcp.custom_route("/api/v1/activities", methods=["POST"])
    async def api_create_activity(request):
        """Create a new activity."""
        if err := auth.require_auth(request):
            return err
        try:
            data = CreateActivityInput(**(await request.json()))
        except ValidationError as e:
            return pydantic_error(e)
        result = await asyncio.to_thread(
            db.add_activity,
            data.case_id,
            data.description,
            data.activity_type,
            data.date,
            data.minutes,
        )
        return JSONResponse({"success": True, "activity": result})

    @mcp.custom_route("/api/v1/activities/{activity_id}", methods=["DELETE"])
    async def api_delete_activity(request):
        """Delete an activity."""
        if err := auth.require_auth(request):
            return err
        activity_id = int(request.path_params["activity_id"])
        deleted = await asyncio.to_thread(db.delete_activity, activity_id)
        if deleted:
            return JSONResponse({"success": True})
        return api_error("Activity not found", "NOT_FOUND", 404)

File: frontend/src/api/activities.ts
import type { Activity, CreateActivityInput } from '../types';
import { request } from './common';

export async function createActivity(
  input: CreateActivityInput
): Promise<{ success: boolean; activity: Activity }> {
  return request('/activities', {
    method: 'POST',
    body: JSON.stringify(input),
  });
}

export async function deleteActivity(activityId: number): Promise<{ success: boolean }> {
  return request(`/activities/${activityId}`, {
    method: 'DELETE',
  });
}
